samples/Identity/src/main/java/identity/FleetProvisioningSample.java

```diff
@@ -75,37 +77,42 @@ static void onRejectedRegister(ErrorResponse response) {
                 ", statusCode: " + response.statusCode);
 
         gotResponse.complete(null);
-        System.exit(1);
     }
 
     static void onCreateKeysAndCertificateAccepted(CreateKeysAndCertificateResponse response) {
-        System.out.println("CreateKeysAndCertificate response certificateId: " + response.certificateId);
         if (response != null) {
-            createKeysAndCertificateResponse = response;
+            System.out.println("CreateKeysAndCertificate response certificateId: " + response.certificateId);
+            if (createKeysAndCertificateResponse == null) {
+                createKeysAndCertificateResponse = response;
+            }
         } else {
             System.out.println("CreateKeysAndCertificate response is null");
         }
         gotResponse.complete(null);
     }
 
     static void onCreateCertificateFromCsrResponseAccepted(CreateCertificateFromCsrResponse response) {
-        System.out.println("CreateCertificateFromCsr response certificateId: " + response.certificateId);
         if (response != null) {
-            createCertificateFromCsrResponse = response;
+            System.out.println("CreateCertificateFromCsr response certificateId: " + response.certificateId);
+            if (createCertificateFromCsrResponse == null) {
```

Comment:
Is it possible for createCertificateFromCsrResponse to not be null? And in that event, should we log that we encountered that scenario with an else { print }?

Reply:
It shouldn't be possible, but a customer suggested that maybe they are sending two requests and that is causing the issue. I'll add an `else` print, better to have more information than not enough 👍